sdk/client/src/client.rs

```diff
@@ -82,32 +82,47 @@ impl Client {
 
         let start = std::time::Instant::now();
         while start.elapsed() < timeout.unwrap_or(DEFAULT_TIMEOUT) {
-            let txn_resp = self
-                .get_account_transaction(address, seq, true)
-                .await
-                .map_err(WaitForTransactionError::GetTransactionError)?;
-            if let (Some(txn), state) = txn_resp.into_parts() {
-                if txn.hash.to_hex() != txn_hash {
-                    return Err(WaitForTransactionError::TransactionHashMismatchError(txn));
-                }
-                match txn.vm_status {
-                    diem_json_rpc_types::views::VMStatusView::Executed => {}
-                    _ => return Err(WaitForTransactionError::TransactionExecutionFailed(txn)),
-                }
-                return Ok(Response::new(txn, state));
-            }
-
-            if let Some(state) = self.last_known_state() {
-                if expiration_time_secs <= state.timestamp_usecs / 1_000_000 {
-                    return Err(WaitForTransactionError::TransactionExpired);
-                }
+            if let Some(res) = self.check_account_transaction_exists(address, seq, expiration_time_secs, txn_hash).await {
```

Comment:
Since we're dropping the `wait_for` functions lets revert this change, and the one below.

Reply:
Would you be able to revert the changes to this file?

Reply:
yes, sorry about the confusion